modules/configuration/src/testFixtures/java/org/apache/ignite/internal/configuration/testframework/ConfigurationExtension.java

```diff
@@ -84,58 +91,100 @@ public class ConfigurationExtension implements BeforeEachCallback, AfterEachCall
         context.getStore(NAMESPACE).put(POOL_KEY, newSingleThreadExecutor());
+
+        injectFields(context, true);
     }
 
     /** {@inheritDoc} */
     @Override
     public void afterAll(ExtensionContext context) throws Exception {
-        ExecutorService pool = context.getStore(NAMESPACE).remove(POOL_KEY, ExecutorService.class);
+        context.getStore(NAMESPACE).remove(CGEN_KEY);
 
-        pool.shutdownNow();
+        context.getStore(NAMESPACE).remove(POOL_KEY, ExecutorService.class).shutdownNow();
+
+        context.getStore(NAMESPACE).remove(REVISION_LISTENER_HOLDER_KEY);
     }
 
     /** {@inheritDoc} */
     @Override
     public void beforeEach(ExtensionContext context) throws Exception {
-        ConfigurationAsmGenerator cgen = new ConfigurationAsmGenerator();
+        injectFields(context, false);
+    }
 
-        context.getStore(NAMESPACE).put(CGEN_KEY, cgen);
+    private void injectFields(ExtensionContext context, boolean forStatic) throws Exception {
+        Class<?> testClass = context.getRequiredTestClass();
+        Object testInstance = context.getTestInstance().orElse(null);
 
-        Object testInstance = context.getRequiredTestInstance();
+        assert forStatic || testInstance != null;
 
-        ExecutorService pool = context.getStore(NAMESPACE).get(POOL_KEY, ExecutorService.class);
+        Store store = context.getStore(NAMESPACE);
+
+        ConfigurationAsmGenerator cgen = store.get(CGEN_KEY, ConfigurationAsmGenerator.class);
+        ExecutorService pool = store.get(POOL_KEY, ExecutorService.class);
 
         StorageRevisionListenerHolderImpl revisionListenerHolder = new StorageRevisionListenerHolderImpl();
 
-        context.getStore(NAMESPACE).put(REVISION_LISTENER_HOLDER_KEY, revisionListenerHolder);
+        if (forStatic) {
+            store.put(REVISION_LISTENER_HOLDER_KEY, revisionListenerHolder);
+        } else {
+            store.put(REVISION_LISTENER_PER_TEST_HOLDER_KEY, revisionListenerHolder);
+        }
 
-        for (Field field : getInjectConfigurationFields(testInstance.getClass())) {
+        for (Field field : getInjectConfigurationFields(testClass, forStatic)) {
             field.setAccessible(true);
 
             InjectConfiguration annotation = field.getAnnotation(InjectConfiguration.class);
 
-            field.set(testInstance, cfgValue(field.getType(), annotation, cgen, pool, revisionListenerHolder));
+            Object cfgValue = cfgValue(field.getType(), annotation, cgen, pool, revisionListenerHolder);
+
+            field.set(forStatic ? null : testInstance, cfgValue);
         }
 
-        for (Field field : getInjectRevisionListenerHolderFields(testInstance.getClass())) {
+        for (Field field : getInjectRevisionListenerHolderFields(testClass, forStatic)) {
             field.setAccessible(true);
 
-            field.set(testInstance, revisionListenerHolder);
+            field.set(forStatic ? null : testInstance, revisionListenerHolder);
```

Comment:
Maybe skip this cycle for static fields?

Reply:
Why? We need to inject static fields

Reply:
This is a static field and does not need an instance, my mistake.